rclcpp_components/cmake/rclcpp_components_register_node.cmake

```diff
@@ -22,15 +22,24 @@
 # :type target: string
 # :param PLUGIN: the plugin name
 # :type PLUGIN: string
-# :type EXECUTABLE: the node's executable name
+# :param EXECUTABLE: the node's executable name
 # :type EXECUTABLE: string
+# :param RESOURCE_INDEX: the ament resource index to register the components
+# :type RESOURCE_INDEX: string
 #
 macro(rclcpp_components_register_node target)
-  cmake_parse_arguments(ARGS "" "PLUGIN;EXECUTABLE" "" ${ARGN})
+  cmake_parse_arguments(ARGS "" "PLUGIN;EXECUTABLE;RESOURCE_INDEX" "" ${ARGN})
   if(ARGS_UNPARSED_ARGUMENTS)
     message(FATAL_ERROR "rclcpp_components_register_node() called with unused "
       "arguments: ${ARGS_UNPARSED_ARGUMENTS}")
   endif()
+  # default to rclcpp_components if not specified otherwise
+  set(resource_index "rclcpp_components")
+  if(NOT "${ARGS_RESOURCE_INDEX}" STREQUAL "")
+    set(resource_index ${ARGS_RESOURCE_INDEX})
+    message(STATUS "Setting component resource index to non-default value ${resource_index}")
+  endif()
+
   set(component ${ARGS_PLUGIN})
```

Comment:
I didn't want to pollute the changes any further, but I think the two args (`PLUGIN` and `EXECUTABLE`) could be checked for empty strings.

Reply:
Feel free to add it, or we can address it in a follow-up Issue/PR.